test/compress/issue-1443.js

```diff
@@ -0,0 +1,53 @@
+unsafe_undefined: {
+    options = {
+        comparisons: true,
+        unsafe: true
+    }
+    mangle = {}
+    input: {
+        function f(undefined) {
+            var a;
+            return function() {
+                console.log(typeof a == "undefined");
+            }
+        }
+    }
+    expect: {
+        function f(n) {
+            var o;
+            return function() {
+                console.log(n===o);
+            }
+        }
+    }
+}
+
+keep_fnames: {
+    options = {
+        comparisons: true,
+        unsafe: true
+    }
+    mangle = {
+        keep_fnames: true
+    }
+    input: {
+        function f(undefined) {
+            return function() {
+                function n(a) {
+                    return a * a;
+                }
+                return typeof n == "undefined";
+            };
+        }
+    }
+    expect: {
+        function f(r) {
+            return function() {
+                function n(n) {
+                    return n * n;
+                }
+                return r === n;
```

Comment:
In light of the fact that `undefined` is a pseudo variable that can be overridden in the parameter of `f(undefined)` is this expected result just accidentally correct because a local function will not triple equal anything passed into the outer `function f`? Had `n` in the original input code been an AST_Constant it could produce incorrect results.

I think this why that specific transform you mention above is in `unsafe`. It assumes that `undefined` is not overwritten.